src/administration/install_docker.md

```diff
@@ -1,78 +1,118 @@
 
-## Reverse Proxy / Webserver
+Customise both the `docker-compose.yml` and the Lemmy [configuration file](configuration.md) `lemmy.hjson` to suit the environment in question.
 
-Here's an optional nginx reverse proxy template, which you can place in `/etc/nginx/sites-enabled`
+#### Database tweaks (optional);
 
-Alternatively, you can use any other web server such as caddy as a simple reverse proxy.
+It is possible to optimise the postgres configuration for the expected workload and hardware configuration; the following website can be used to generate a customised `postgresql.conf` file: <https://pgtune.leopard.in.ua/>
 
-Be sure to edit the `{{ }}` to match your domain and port.
+An example configuration file is also available from the `lemmy-ansible` repository:
+
+```bash
+pushd config
+curl -O https://raw.githubusercontent.com/LemmyNet/lemmy-ansible/main/examples/customPostgresql.conf
+popd
+```
 
-If you're using this, you will need to set up Let's Encrypt. See those instructions below.
+### TLS certificates
 
-`wget https://raw.githubusercontent.com/LemmyNet/lemmy-ansible/main/templates/nginx.conf`
+In 2023 the use of TLS is Mandatory. [Let's Encrypt](https://letsencrypt.org/) can be used to generate TLS certificates for the reverse proxy.
 
-If you've set up Let's Encrypt and your reverse proxy, you can go to `https://{{ domain }}`
+Here's a guide for accomplishing this [on Ubuntu](https://www.digitalocean.com/community/tutorials/how-to-secure-nginx-with-let-s-encrypt-on-ubuntu-22-04).
 
-## Let's Encrypt
+Traefik can [automate the entire process](https://doc.traefik.io/traefik/https/acme/).
 
-You should also setup TLS, for example with [Let's Encrypt](https://letsencrypt.org/). [Here's a guide for setting up letsencrypt on Ubuntu](https://www.digitalocean.com/community/tutorials/how-to-secure-nginx-with-let-s-encrypt-on-ubuntu-22-04).
+For federation to work, it is important that the reverse proxy does not change any headers that form part of the `signature`. This includes the `Host` header; refer to the documentation for the reverse proxy if it is not NGINX or Traefik to ensure that it is configured to pass the `Host` header unmodified.
 
-For federation to work, it is important that you do not change any headers that form part of the signature. This includes the `Host` header - you may need to refer to the documentation for your proxy server to pass through the `Host` header unmodified.
+## Deployment
 
-# Updating
+Run the following commands to deploy the Lemmy stack:
 
-To update to the newest version, you can manually change the version in `docker-compose.yml`. Alternatively, fetch the latest version from our [lemmy-ansible](https://github.com/LemmyNet/lemmy-ansible) repo:
+`docker-compose`:
 
 ```bash
-wget https://raw.githubusercontent.com/LemmyNet/lemmy-ansible/main/templates/docker-compose.yml
-# Then replace the {{ }} vars again
 docker-compose up -d
 ```
+
+`docker stack`:
```

Comment:
docker stack? what is this? That isn't how to deploy lemmy.

Reply:
`docker stack` is Docker's container orchestration tool for deployment to multi-node docker clusters. It provides better tooling to enable high availability and horizontal scaling. `docker compose` is for testing your multi-container deployments; stack is

Reply:
docker stack is still an appropriate mechanism to deploy non-scaled services with the benefit that it each service can migrate within the swarm nodes in the event of e.g. a power failure. It should be documented for completeness sake, even if compose is the typical / preferred option.